trail_detection_node/trail_detection_node/v2_trailDetectionNode.py

```diff
@@ -0,0 +1,256 @@
+import torch
+from .model_loader import PSPNet
+from PIL import Image as ImagePIL
+from torchvision import transforms
+import cv2
+
+import rclpy
+from rclpy.node import Node
+from geometry_msgs.msg import PoseStamped
+from sensor_msgs.msg import Image, PointCloud2
+import sensor_msgs_py.point_cloud2 as pc2
+import numpy as np
+import math
+from cv_bridge import CvBridge
+
+import message_filters
+
+'''
+    The transformation matrix as well as the coordinate conversion and depth estimation functions are copied from human_detection_node
+'''
+camera_transformation_k = np.array([
+    [628.5359544,0,676.9575694],
+    [0,627.7249542,532.7206716],
+    [0,0,1]])
+
+rotation_matrix = np.array([
+    [-0.007495781893,-0.0006277316155,0.9999717092],
+    [-0.9999516401,-0.006361853422,-0.007499625104],
+    [0.006366381192,-0.9999795662,-0.0005800141927]])
+
+rotation_matrix = rotation_matrix.T
+
+translation_vector = np.array([-0.06024059837, -0.08180891509, -0.3117851288])
+image_width=1280
+image_height=1024
+
+def convert_to_lidar_frame(uv_coordinate):
+    """
```

Comment:
is this code all copied from nathan's node? It's ok but leave a comment so we know to make any changes to both (instead of rewriting both to use these functions from one location).